src/App.js

```diff
@@ -1,25 +1,83 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from "react";
+import "bootstrap/dist/css/bootstrap.css";
+import "./App.css";
+import ContactList from "./ContactList";
+import { Route, Switch } from "react-router-dom";
+import ContactProfile from "./ContactProfile";
+import AddContact from "./AddContact";
+import EditContact from "./EditContact";
+
+export const ContactContext = React.createContext();
 
 function App() {
+  const [contacts, setContacts] = useState(contactsApi.contacts);
+
+  const contactContextValue = {
+    contacts: contacts,
+    handleContactAdd,
+    handleContactDelete,
+    handleContactEdit,
+  };
+
+  function handleContactAdd(newContact) {
+    const newContactsList = [...contacts, newContact];
+    setContacts(newContactsList);
+  }
+
+  function handleContactDelete(id) {
+    const newContactList = contacts.filter((c) => c.id !== id);
+    setContacts(newContactList);
+  }
+
+  function handleContactEdit(
+    id,
+    editedName,
+    editedImage_url,
+    editedEmail,
+    editedPhone_number
+  ) {
+    const contactList = [...contacts];
+    const indexToEdit = contactList.findIndex((c) => c.id === id);
+    const editedContact = {
+      id,
+      name: editedName,
+      image_url: editedImage_url,
+      email: editedEmail,
+      phone_number: editedPhone_number,
+    };
+    contactList[indexToEdit] = editedContact;
```

Comment:
This is okay, but it assumes that you'll pass it all 5 values. If you were to instead be able to pass in a single object argument (which would represent the newly edited contact) and it was called `editedContact`, then something better might be:

```js
setContacts(contacts.map((contact, i) => {
  if (i !== indexToEdit) {
    return contact
  }

  return {
    ...contact,
    ...editedContact
  }
}))
```